mongo/client_test.go

```diff
@@ -344,4 +344,49 @@ func TestClient(t *testing.T) {
 			})
 		}
 	})
+	t.Run("serverAPI version", func(t *testing.T) {
+		t.Run("success", func(t *testing.T) {
+			ServerAPIOptions := options.ServerAPI().SetServerAPIVersion(options.ServerAPIVersion1)
+			client, err := NewClient(&options.ClientOptions{ServerAPIOptions: ServerAPIOptions})
+			assert.Nil(t, err, "unexpected error from NewClient: %v", err)
+			assert.Equal(t, ServerAPIOptions, client.serverAPI,
+				"mismatch in serverAPI; expected %v, got %v", ServerAPIOptions, client.serverAPI)
+		})
+		t.Run("success with more options", func(t *testing.T) {
```

Comment:
I think it's good to make tests small and self-contained, but the test above ("success") is a subset of this one. I think we can remove the "success" test and rename this one to something like "success with all options" without losing any functionality or readability.

Reply:
Sounds good, gone.